modules/request/main/index.coffee

```diff
@@ -1,13 +1,9 @@
-hasResponse = (request) ->
-  responseType = request.responseType
-  if responseType and responseType isnt 'text' then request.response else request.responseText
```

Comment:
This file probably needs looking over just to make sure it's still sensible. The changes made are:
- Remove support for old browsers (XDomainRequest and onreadystatechange)
- Remove support for request responses that have content but no status as this should not be possible
- Restructure reshaped request to output less code (and make 100% coverage possible)